File: frontend/src/api/ai-call-log.ts
// AI 调用日志接口文件：封装日志列表与详情查询请求。
import http from './http'

// AI 调用日志列表查询参数
export interface AiCallLogListParams {
  module?: string
  task_type?: string
  status?: string
  keyword?: string
  start_time?: string
  end_time?: string
  page?: number
  page_size?: number
}

// AI 调用日志列表项类型
export interface AiCallLogItem {
  id: number
  module: string | null
  task_type: string | null
  prompt_template_key: string | null
  prompt_version: string | null
  prompt: string
  response: string | null
  status: string
  error_message: string | null
  model_name: string
  latency_ms: number | null
  created_at: string
}

// AI 调用日志分页结果类型
export interface AiCallLogListResult {
  total: number
  page: number
  page_size: number
  items: AiCallLogItem[]
}

// AI 调用日志详情类型：详情接口返回完整 prompt、response 和错误信息。
export type AiCallLogDetail = AiCallLogItem

// AI 效果统计类型：用于日志页顶部统计卡片。
export interface AiCallLogSummary {
  total_calls: number
  success_calls: number
  failed_calls: number
  success_rate: number
  avg_latency_ms: number
  customer_follow_advice_calls: number
  customer_ai_adopted_count: number
  customer_ai_adoption_rate: number
}

// Prompt 模板维度调用效果统计类型。
export interface PromptTemplateSummaryItem {
  prompt_template_key: string
  prompt_version: string | null
  module: string | null
  task_type: string | null
  total_calls: number
  success_calls: number
  failed_calls: number
  success_rate: number
  avg_latency_ms: number
}

// 查询 AI 调用日志列表。
export const aiCallLogListApi = (params: AiCallLogListParams) => {
  return http.get('/api/ai-call-logs', { params })
}

// 查询 AI 调用日志详情。
export const aiCallLogDetailApi = (id: number) => {
  return http.get(`/api/ai-call-logs/${id}`)
}

// 查询 AI 效果统计。
export const aiCallLogSummaryApi = () => {
  return http.get('/api/ai-call-logs/summary')
}

// 查询 Prompt 模板维度调用效果统计。
export const aiCallLogPromptSummaryApi = () => {
  return http.get('/api/ai-call-logs/prompt-summary')
}

File: backend/app/schemas/ai_call_log.py
"""AI 调用日志 Schema 文件：定义日志查询接口的返回结构。"""

from pydantic import BaseModel


class AiCallLogListItem(BaseModel):
    """AI 调用日志列表项：列表页只返回摘要字段，避免长文本影响响应体积。"""

    id: int
    module: str | None = None
    task_type: str | None = None
    prompt: str
    response: str | None = None
    status: str
    error_message: str | None = None
    model_name: str
    latency_ms: int | None = None
    created_at: str


class AiCallLogDetail(BaseModel):
    """AI 调用日志详情：返回完整 prompt、response 和错误信息。"""

    id: int
    module: str | None = None
    task_type: str | None = None
    prompt: str
    response: str | None = None
    status: str
    error_message: str | None = None
    model_name: str
    latency_ms: int | None = None
    created_at: str


class AiCallLogListResult(BaseModel):
    """AI 调用日志分页结果：用于承载总数、分页参数和列表数据。"""

    total: int
    page: int
    page_size: int
    items: list[AiCallLogListItem]
